### DISCUSSION TITLE
Where can i find Performance metrics of Edgex Foundry

### DISCUSSION THREAD

**Discussion by kumardeepak5** [edgexfoundry/community#201] [Q&A]
Hi Team,

I am trying to find performance report of Edgex Foundry (minnesota release).

Basically i am looking for the answer of below queries- (please consider Edgex foundry 3.0 with default configuration)

- Maximum number of devices could be integrated with single REST Service.
- At one time how many devices data can be caputured using edgex

Thanks

**Comment by jiekechoo** [comment]
@kumardeepak5 
Welcome, it's a good question.
In my experience, it depends on your `Hardward` configuration.

In each EdgeX micro service, there is a `configuration.yaml` file, contains some default config value, see here: https://docs.edgexfoundry.org/3.1/microservices/configuration/CommonConfiguration/ . You could use the command line like below to check yours:
```
curl http://localhost:59880/api/v3/config
```
Q: Maximum number of devices could be integrated with single REST Service.
In my use case, cloud VM with: 4 cores, 8G mem, it can handle 50000-80000 `virtual` devices. 

Q: At one time how many devices data can be caputured using edgex
How many `commands` and what is the command `interval` in your test? And, there are many options you should concerned, such as: data persist, southbound device protocols.

Maybe, these are helpful.

Jieke
YIQISOFT

**Comment by JamesKButcher** [comment]
Hi Deepak,

Yes a glossy performance report like we did for Jakarta and Kamakura etc was never completed for the Minnesota release, although the raw information is here: https://wiki.edgexfoundry.org/display/FA/Minnesota+Data

We will certainly do one for the Napa release when we have the raw data. Note the Napa release is due this week (Nov 15th all going well).

All of that said, I don't think a maximal number of devices is one of the metrics that are captured in these tests, and of course it is dependent on the hardware.

But Jieke has posted some good empirical data above 👍 

Thanks
James

**Comment by JamesKButcher** [comment]
The Napa version of the EdgeX performance metrics is now available here: https://wiki.edgexfoundry.org/display/FA/Napa+Data
